<p>Shared coding conventions allow teams to collaborate effectively. This rule raises an issue when class or interface names are not in
<em>PascalCase</em> (i.e. camel case with an initial capital letter).</p>
<h2>Noncompliant Code Example</h2>
<pre>
class my_class {...}
</pre>
<h2>Compliant Solution</h2>
<pre>
class MyClass {...}
</pre>

