<p>A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many
things.</p>
<h2>Noncompliant Code Example</h2>
<p>With a maximum number of 4 parameters:</p>
<pre>
function doSomething(param1: number, param2: number, param3: number, param4: number, param5: number) {
...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function doSomething(param1: number, param2: number, param3: number, param4: number) {
...
}
</pre>
<h2>Exceptions</h2>
<p>Parameter properties are ignored.</p>
<pre>
class A {
    constructor(public p1: number, public p2: number, public p3: number, public p4: number, public p5: number) {
    }
}
</pre>

