<p>Boolean literals should be avoided in comparison expressions <code>==</code>, <code>!=</code> to improve code readability. This rule also reports
on redundant boolean operations.</p>
<h2>Noncompliant Code Example</h2>
<pre>
if (booleanMethod() == true) { /* ... */ }
if (booleanMethod() == false) { /* ... */ }
if (booleanMethod() || false) { /* ... */ }
doSomething(!false);
doSomething(booleanMethod() == true);

booleanVariable = booleanMethod() ? true : false;
booleanVariable = notBooleanMethod() ? true : false;
</pre>
<h2>Compliant Solution</h2>
<pre>
if (booleanMethod()) { /* ... */ }
if (!booleanMethod()) { /* ... */ }
if (booleanMethod()) { /* ... */ }
doSomething(true);
doSomething(booleanMethod());

booleanVariable = booleanMethod();
booleanVariable = Boolean(notBooleanMethod());
</pre>
<h2>Exceptions</h2>
<p>Expressions with strict comparison operators (<code>===</code> and <code>!==</code>) are ignored.</p>

