<p>Sharing some naming conventions is a key point to make it possible for a team to efficiently collaborate. This rule allows to check that all
variable names match one of the following patterns: <code>lowerCamelCase</code>, <code>PascalCamelCase</code> or <code>UPPER_CASED</code>. This rule
also allows leading underscore.</p>
<h2>Noncompliant Code Example</h2>
<pre>
const foo_bar = 1;
const baz_ = 2;
</pre>
<h2>Compliant Solution</h2>
<pre>
const fooBar = 1;
const _baz = 2;
</pre>

