<p>TypeScript provides the <code>import</code> and <code>export</code> keywords as language-standard mechanisms for module management. Javascript's
<code>require()</code> usages should be converted to the new syntax.</p>
<h2>Noncompliant Code Example</h2>
<pre>
// circle.js
module.exports = function (r) {
  return PI * r * r;
};

// foo.js
const circle = require('./circle.js');  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
// circle.ts
export default function (r) {
  return PI * r * r;
}

// foo.ts
import circle from "./circle.ts"
</pre>

