<p>Jump statements, such as <code>return</code>, <code>break</code> and <code>continue</code> let you change the default flow of program execution,
but jump statements that direct the control flow to the original direction are just a waste of keystrokes.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function redundantJump(x: number) {
  if (x == 1) {
    console.log("x == 1");
    return;
//  ^^^^^^^ {{Remove this redundant jump.}}
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function redundantJump(x: number) {
  if (x == 1) {
    console.log("x == 1");
  }
}
</pre>
<h2>Exceptions</h2>
<p><code>break</code> and <code>return</code> inside <code>switch</code> statement are ignored, because they are often used for consistency.
<code>continue</code> with label is also ignored, because label is usually used for clarity.</p>

