<p>In TypeScript a function can return a value explicitly, by using a <code>return</code> statement with a value, or implicitly, at the end of the
function or by a <code>return</code> with no value, resulting in the function returning <code>undefined</code>. Implicit returns of
<code>undefined</code> not declared in the function signature, can be confusing for the maintainer.</p>
<p>This rule ensures that <code>returns</code> are either all explicit or all implicit, or the function signatures makes the implicit
<code>return</code> obvious.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function foo(a) { // Noncompliant, function exits without "return"
  if (a == 1) {
    return true;
  }
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function foo(a): boolean | undefined {
  if (a == 1) {
    return true;
  }
}
</pre>

