<p>Many of JavaScript's <code>Array</code> methods return an altered version of the array while leaving the source array intact.
<code>Array.reverse()</code> is not one of those. Instead, it alters the source array <em>in addition to</em> returning the altered version, which is
likely not what was intended. </p>
<h2>Noncompliant Code Example</h2>
<pre>
var b = a.reverse(); // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
var b = [...a].reverse();  // de-structure and create a new array, so reverse doesn't impact 'a'

a.reverse();
</pre>

