<p>Variables can be declared with or without types. Variables declared without a type will be implicitly typed if the declaration includes an
initialization, and compiler type checking will be automatically applied to any typed variable. But if you declare a variable with the
<code>any</code> "type" then you've explicitly told the compiler not to do any type checking, which is risky. </p>
<h2>Noncompliant Code Example</h2>
<pre>
let a = 42;  // implicitly typed to number
let b: number = 42;  // explicitly typed to number
let c: any = 42;  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>
let a = 42;
let b: number = 42;
let c: number = 42;
</pre>

