<p>Assigning <code>this</code> to a local variable is a way to reference parent context inside inner functions. In TypeScript when using arrow
functions this happens automatically.</p>
<p>This rule raises an issue when <code>this</code> is assigned to a local variable.</p>
<h2>Noncompliant Code Example</h2>
<pre>
function Foo() {
  let that = this;  // Noncompliant
  that.val = 0;

  setInterval(function() {
    that.val++;
  }, 1000);
}
</pre>
<h2>Compliant Solution</h2>
<pre>
function Foo() {
  this.val = 0;

  setInterval(() =&gt; {
    this.val++;
  }, 1000);
}
</pre>
<h2>Exceptions</h2>
<p>This rule ignores <code>this</code> used for destructuring.</p>
<pre>
const { foo, bar } = this;
</pre>

